### ISSUE TITLE
[BUG] DisjointCNN performs poorly on problems > 2 classes

### ISSUE THREAD

**Issue by TonyBagnall** [aeon-toolkit/aeon#3775]
### Describe the bug

I'm running classifiers on the multiverse core 

https://github.com/aeon-toolkit/multiverse/

and something wrong with DisjopintCNN I think. 

Handwriting (26 classes) scores 0.055 against a 1/26 = 0.038 chance rate, and
PhonemeSpectra (39 classes) scores 0.075 against 1/39 = 0.026.

### Published vs obtained

Published values are Table 3 of the Disjoint-CNN paper; ours are single runs on the
default train/test split with `random_state=0` and otherwise default parameters.

| dataset | published | aeon | difference |
|---|---|---|---|
| ArticularyWordRecognition | 0.984 | 0.277 | -0.708 |
| UWaveGestureLibrary | 0.924 | 0.366 | -0.559 |
| Libras | 0.916 | 0.533 | -0.383 |
| ERing | 0.964 | 0.589 | -0.375 |
| LSST | 0.547 | 0.270 | -0.277 |
| Handwriting | 0.321 | 0.055 | -0.266 |
| Cricket | 0.999 | 0.750 | -0.249 |
| PhonemeSpectra | 0.282 | 0.075 | -0.207 |
| PEMS-SF | 0.817 | 0.642 | -0.176 |
| AtrialFibrillation | 0.375 | 0.200 | -0.175 |
| EigenWorms | 0.593 | 0.420 | -0.174 |
| StandWalkJump | 0.436 | 0.267 | -0.169 |
| SelfRegulationSCP1 | 0.903 | 0.737 | -0.166 |
| Heartbeat | 0.759 | 0.615 | -0.145 |
| NATOPS | 0.967 | 0.828 | -0.139 |
| DuckDuckGeese | 0.644 | 0.520 | -0.124 |
| RacketSports | 0.846 | 0.763 | -0.083 |
| EthanolConcentration | 0.309 | 0.232 | -0.077 |
| MotorImagery | 0.569 | 0.500 | -0.069 |
| HandMovementDirection | 0.324 | 0.257 | -0.068 |
| PenDigits | 0.995 | 0.962 | -0.033 |
| FaceDetection | 0.567 | 0.538 | -0.028 |
| Epilepsy | 0.989 | 0.971 | -0.018 |
| **mean** | **0.697** | **0.494** | **-0.203** |

For context, on the 55 datasets where we have results for several deep classifiers,
DisjointCNN averages 0.606 where H-InceptionTime averages 0.745, LITETime-MV 0.755 and
ConvTran 0.751. It is the only deep learner in that set that finishes below 1NN-DTW.

### Steps/Code to reproduce the bug

```python
from aeon.classification.deep_learning import DisjointCNNClassifier
from aeon.datasets import load_classification

X_train, y_train = load_classification("ERing", split="train")
X_test, y_test = load_classification("ERing", split="test")

clf = DisjointCNNClassifier(random_state=0)
clf.fit(X_train, y_train)
print(clf.score(X_test, y_test))
# obtained around 0.589; the paper reports 0.964
```

ArticularyWordRecognition shows it more starkly (0.277 against a published 0.984) but
takes longer to run.

### Expected results

Accuracy in the region of the authors' published values, and in particular not close to
chance on problems with many classes.

### Actual results

Consistently below the published values on all 23 shared datasets, approaching chance as
the number of classes grows.

   

### Versions

aeon 1.5.0

**Comment by hadifawaz1999** [comment]
Might be because of normalization done in the original code of the paper: https://github.com/Navidfoumani/Disjoint-CNN/blob/38180ccc1f050a3fee6961f5b4ea3bc132016ab3/utils/data_loader.py#L6

**Comment by TonyBagnall** [comment]
I vendored in from the original and claude code thinks this is the bug, I have no idea

_one_plus_one_d_convolution_layer permutes at the end of every block (aeon/networks/encoder/_disjoint_cnn.py#L308). The authors permute only between blocks — DCNN_4L.py has Permute at L32, L42, L52 — after blocks 1–3 — then goes straight from conv4
  to MaxPooling2D at L63.                                                                                                                                                                                                                                 
                                                                                                                                                                                                                                                          
  The trailing permute leaves (time, filters, 1) entering GlobalAveragePooling2D, which reduces the two leading axes — so the head gets 1 feature instead of 64. Dense(128) having 256 params (1×128+128) confirms it. Hence the collapse scaling with    
  class count.                                                                                                                                                                                                                                            
                                                                                                                                                                                                                                                          
  The fix is to skip the permute on the final block. 


It also highlighted four differences 

eparate from the bug, and all places aeon departs from the authors:                                                                                                                                                                                    
                                                                                                                                                                                                                                                          
                                       | authors      original                                   │                aeon
  │ 1. Class-weighted loss │ class_weight=create_class_weight(...), defined here as log(2·total/count) floored at 1 │ none                               │                                                                                                                                                                                            
  │ 2. Batch size          │ min(n_cases/10, batch_size) with batch_size = 8 (1 for EigenWorms/DuckDuck, L173)      │ flat 16; use_mini_batch_size=False │                                                                                                        
  │ 3. Epochs              │ 500                                                                                    │ 2000                               │                                                                                                
                                                                                    
  │ 4. What's monitored    │ val_loss for both ReduceLROnPlateau and ModelCheckpoint                                │ "loss" (L337, L341)                │

**Comment by TonyBagnall** [comment]
not run a full test, so above is indicative

aeon permutes after every block including the last, where the authors permute only between blocks. Decisive test — add an inverse permute on the final block only, changing nothing else in aeon:

The permute alone takes aeon from 0.641 → 0.941. That settles it. Let me pull exact line numbers and a pinned commit for the links:

**Comment by TonyBagnall** [comment]
I'll run the vendored version if ts2vec ever finishes, 60 hours on an H200 on one problem is some compute

**Comment by hadifawaz1999** [comment]
you're right, i just noticed that in my deep-rehab-pile code i dont permute the last block, so its my aeon version that was wrong.... https://github.com/MSD-IRIMAS/DeepRehabPile/blob/main/deep_rehab_pile/classifiers/_discnn.py

Good catch !
